src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,221 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+function gpkgread(fname; layer=1)
+  db = SQLite.DB(fname)
+  assertgpkg(db)
+  table, geoms = gpkgtable(db; layer)
+  DBInterface.close!(db)
+  georef(table, geoms)
+end
+
+function assertgpkg(db)
+  # According to https://www.geopackage.org/spec/#r6 and https://www.geopackage.org/spec/#r7
+  # PRAGMA integrity_check returns a single row with the value 'ok'
+  # PRAGMA foreign_key_check (w/ no parameter value) returns an empty result set
+  if first(DBInterface.execute(db, "PRAGMA integrity_check;")).integrity_check != "ok" ||
+     !(isempty(DBInterface.execute(db, "PRAGMA foreign_key_check;")))
+    throw(ErrorException("database integrity at risk or foreign key violation(s)"))
+  end
+  # According to https://www.geopackage.org/spec/#r10 and https://www.geopackage.org/spec/#r13
+  # A GeoPackage SHALL include a 'gpkg_spatial_ref_sys' table and a 'gpkg_contents table'
+  if first(DBInterface.execute(
+    db,
+    """
+  SELECT COUNT(*) AS n FROM sqlite_master WHERE 
+  name IN ('gpkg_spatial_ref_sys', 'gpkg_contents') AND 
+  type IN ('table', 'view');
+    """
+  )).n != 2
+    throw(ErrorException("missing required metadata tables in the GeoPackage SQL database"))
+  end
+end
+
+# According to Geometry Columns Table Requirements
+# https://www.geopackage.org/spec/#:~:text=2.1.5.1.2.%20Table%20Data%20Values
+#------------------------------------------------------------------------------
+# Requirement 22: gpkg_geometry_columns table
+# SHALL contain one row record for the geometry column
+# in each vector feature data table
+#
+# Requirement 23: gpkg_geometry_columns table_name column
+# SHALL reference values in the gpkg_contents table_name column
+# for rows with a data_type of 'features'
+#
+# Requirement 24: The column_name column value in a gpkg_geometry_columns row
+# SHALL be the name of a column in the table or view specified by the table_name
+# column value for that row.
+#
+# Requirement 25: The geometry_type_name value in a gpkg_geometry_columns row
+# SHALL be one of the uppercase geometry type names specified
+
+# Requirement 26: The srs_id value in a gpkg_geometry_columns table row
+# SHALL be an srs_id column value from the gpkg_spatial_ref_sys table.
+#
+# Requirement 27: The z value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 28: The m value in a gpkg_geometry_columns table row SHALL be one
+# of 0, 1, or 2.
+#
+# Requirement 146: The srs_id value in a gpkg_geometry_columns table row
+# SHALL match the srs_id column value from the corresponding row in the
+# gpkg_contents table.
+function gpkgtable(db, ; layer=1)
+  resultrows = DBInterface.execute(
```

Comment:
DBInterface.execute always returns a Tables.jl table? If this object is a valid table, we should name it `rowtable` to make it clear that we can use the Tables.jl api to query rows, columns ,etc.